src/Altinn.Notifications.Integrations/SendCondition/SendConditionClient.cs

```diff
@@ -0,0 +1,54 @@
+﻿using System.Text.Json;
+
+using Altinn.Notifications.Core.Integrations;
+using Altinn.Notifications.Core.Models.SendCondition;
+using Altinn.Notifications.Core.Shared;
+
+namespace Altinn.Notifications.Integrations.SendCondition
+{
+    /// <summary>
+    /// Implementation of <see cref="IConditionClient"/> using a maskinporten client with Digdir credentials
+    /// </summary>
+    public class SendConditionClient : IConditionClient
+    {
+        private readonly HttpClient _client;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="SendConditionClient"/> class.
+        /// </summary>
+        public SendConditionClient(HttpClient client)
+        {
+            _client = client;
+        }
+
+        /// <inheritdoc/>
+        public async Task<Result<bool, ConditionClientError>> CheckSendCondition(Uri url)
+        {
+            HttpResponseMessage res = await _client.GetAsync(url);
+            string responseString = await res.Content.ReadAsStringAsync();
+
+            if (res.IsSuccessStatusCode)
+            {
+                try
+                {
+                    SendConditionResponse? conditionResponse = JsonSerializer.Deserialize<SendConditionResponse?>(responseString);
+
+                    if (conditionResponse?.SendNotification != null)
+                    {
+                        return (bool)conditionResponse.SendNotification;
```

Comment:
I'm getting an error if I try to cast null to a bool. Might need to add a check.

Reply:
Isn't the check already there? line 36

Reply:
Sorry, just me being blind. In that case the cast is unnecessary. 
```suggestion
                        return conditionResponse.SendNotification.Value;
```

Reply:
Looks like the compiler produce the same code in both cases. I still prefer to avoid casting when I can.